lib/src/frontend/spawn_hybrid.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:async/async.dart';
+import 'package:path/path.dart' as p;
+import 'package:stream_channel/stream_channel.dart';
+
+import '../backend/invoker.dart';
+import '../util/remote_exception.dart';
+import '../utils.dart';
+
+/// A transformer that handles messages from the spawned isolate and ensures
+/// that messages sent to it are JSON-encodable.
+final _transformer = new StreamChannelTransformer(
+    new StreamTransformer.fromHandlers(handleData: (message, sink) {
+      switch (message['type']) {
+        case 'data':
+          sink.add(message['data']);
+          break;
+
+        case 'print':
+          print(message['line']);
+          break;
+
+        case 'error':
+          var error = RemoteException.deserialize(message['error']);
+          sink.addError(error.error, error.stackTrace);
+          break;
+      }
+    }), new StreamSinkTransformer.fromHandlers(handleData: (message, sink) {
+      ensureJsonEncodable(message);
+      sink.add(message);
+    }));
+
+/// Spawns a VM isolate for the given [uri], which may be a [Uri] or a [String].
```

Comment:
More specifically, in practice URLs are represented as Strings and Uri objects with similar frequency in Dart. I try to make my APIs handle either representation to avoid making my users write lots of annoying `.toString()` or `Uri.parse()` calls.